src/main/java/org/shanerx/tradeshop/utils/Utils.java

```diff
@@ -476,55 +478,66 @@ public ExchangeStatus canExchangeAll(Shop shop, Inventory playerInv, int multipl
         return ExchangeStatus.SUCCESS; //Successfully completed trade
 	}
 
-    //Returns an arraylist of the itemstacks to be removed/added, if it could not get enough of an item, will return index 0 as null and index 1 as item it could not get enough of
-	public ArrayList<ItemStack> getItems(Inventory inventory, List<ShopItemStack> items, int multiplier) {
-        Inventory clone = Bukkit.createInventory(null, inventory.getStorageContents().length);
-        clone.setContents(inventory.getStorageContents());
-        ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
-        int totalCount = 0, currentCount = 0;
-        debugger.log("ShopTradeListener > Inventory Type Being Searched: " + inventory.getType().name(), DebugLevels.TRADE);
-		debugger.log("ShopTradeListener > Inventory Location Being Searched: " + (inventory.getLocation() != null ? inventory.getLocation().toString() : "null"), DebugLevels.TRADE);
-
-		for (ShopItemStack item : items) {
-			totalCount += item.getAmount() * multiplier;
-			int count = item.getAmount() * multiplier;
+	//Returns an arraylist of the itemstacks to be removed/added, if it could not get enough of an item, will return index 0 as null and index 1 as item it could not get enough of
+	public ArrayList<ItemStack> getItems(Inventory inventory, List<ShopItemStack> search, int multiplier) {
+		Map<ItemStack, Integer> storage = new HashMap<>(), found = new HashMap<>();
+		ArrayList<ItemStack> good = new ArrayList<ItemStack>(), bad = new ArrayList<ItemStack>();
 
-			debugger.log("ShopTradeListener > Item Material Being Searched for: " + item.getItemStack().getType().name(), DebugLevels.TRADE);
-			debugger.log("ShopTradeListener > Item count: " + count, DebugLevels.TRADE);
+		debugger.log("ShopTradeListener > Search List: " + search, DebugLevels.TRADE);
 
-			for (ItemStack storageItem : clone.getStorageContents()) {
-				if (storageItem != null) {
-					boolean isSimilar = item.isSimilar(storageItem.clone());
-					if (isSimilar) {
+		bad.add(null);
 
-						int traded = Math.min(Math.min(storageItem.getAmount(), item.getItemStack().getMaxStackSize()), count);
+		for (ItemStack itemStack : inventory.getStorageContents()) {
+			if (itemStack != null) {
+				ItemStack tempItemStack = itemStack.clone();
+				int amount = tempItemStack.getAmount();
+				tempItemStack.setAmount(1);
```

Comment:
what is this doing?

Reply:
Makes sure that the item-stacks used as keys all have an amount of 1. see my comment on line 483